src/doc-utils.js

```diff
@@ -126,11 +126,19 @@ function propagateBreaks(doc) {
   );
 }
 
+function getNodeSource (node, text) {
```

Comment:
We don't want to make this a generic function, especially given that it is giving the entire line and not the actual range.